src/utils/githubApi.ts

```diff
@@ -202,186 +167,3 @@ export async function getFile(
-}
-
-/**
- * Downloads the entire repository contents of the tag
- *
- * The contents are compressed into a tarball and returned in a buffer that can
- * be streamed for extraction.
- *
- * @param owner Repository owner
- * @param repo Repository name
- * @param sha Revision SHA identifier
- * @returns The tarball data as stream
- */
-export async function downloadSources(
-  owner: string,
-  repo: string,
-  sha: string
-): Promise<NodeJS.ReadableStream> {
-  logger.info(`Downloading sources for ${owner}/${repo}:${sha}`);
-  // TODO add api token to allow downloading from private repos
-  const url = `https://github.com/${owner}/${repo}/archive/${sha}.tar.gz`;
-  const response = await fetch(url);
-  if (!response.ok) {
-    throw new Error(
-      `Unexpected HTTP response from ${url}: ${response.status} (${response.statusText})`
-    );
-  }
-  return response.body;
-}
-
-/**
- * Checks that the HTTP status code matches one of the patterns
- *
- * A pattern can be either a number or a regular expression.
- *
- * @param code HTTP code to check
- * @param patterns A list of patterns to test
- */
-export function codeMatches(
-  code: number,
-  patterns: RetryCodePattern | RetryCodePattern[]
-): boolean {
-  const patternList: RetryCodePattern[] = Array.isArray(patterns)
-    ? patterns
-    : [patterns];
-
-  const stringCode = code.toString();
-  if (stringCode.length !== 3) {
-    return false;
-  }
-
-  for (const pattern of patternList) {
-    if (typeof pattern === 'number') {
-      if (pattern === code) {
-        return true;
-      }
-    } else {
-      // pattern is RegEx
-      if (pattern.test(stringCode)) {
-        return true;
-      }
-    }
-  }
-  return false;
-}
-
-/**
- * Runs the provided function in the retry loop
- *
- * Used for retrying API requests to GitHub
- * NOTE: Octokit now has the retry plugin: https://github.com/octokit/plugin-retry.js
- * So we can probably remove this competely after some time.
- *
- * @param fn Function to retry
- * @param retryParams Retry parameters
- */
-export async function retryHttp<T>(
-  fn: () => Promise<T>,
-  retryParams: Partial<RetryParams> = {}
-): Promise<T> {
-  const params = { ...defaultRetryParams, ...retryParams };
-  const maxRetries = params.retries;
-  let retryNum = 0;
-
-  while (true) {
-    logger.debug(`Retry number ${retryNum}, max retries: ${maxRetries}`);
-
-    try {
-      const result = await fn();
-      return result;
-    } catch (e) {
-      const status = e.status as number;
-      if (params.retryCodes.indexOf(status) > -1 && retryNum < maxRetries) {
-        if (params.cleanupFn) {
-          await params.cleanupFn();
-        }
-        await sleep(params.cooldown);
-        retryNum += 1;
-        continue;
-      } else {
-        if (maxRetries > 0 && retryNum >= maxRetries) {
-          logger.error(`Maximum retries reached (${maxRetries})`);
-        }
-        logger.error(`Error code: ${status}`);
-        throw e;
-      }
-    }
-  }
-}
```

Comment:
Looks like this was not actually used _anywhere_. Eh?

Reply:
It was used in a bunch of places but was obsolete and even erroneous as we already had https://yarnpkg.com/package/@octokit/plugin-retry installed.

Reply:
Gotcha, sorry. Not sure how I missed it but seeing now that we removed three usages from `targets/github.ts`. 👍